components/brave_news/browser/initialization_promise_unittest.cc

```diff
@@ -0,0 +1,138 @@
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/brave_news/browser/initialization_promise.h"
+
+#include <string>
+
+#include "base/functional/bind.h"
+#include "base/functional/callback_helpers.h"
+#include "base/run_loop.h"
+#include "base/test/bind.h"
+#include "brave/components/api_request_helper/api_request_helper.h"
+#include "brave/components/brave_news/browser/brave_news_pref_manager.h"
+#include "brave/components/brave_news/browser/publishers_controller.h"
+#include "brave/components/brave_news/browser/urls.h"
+#include "brave/components/brave_news/common/brave_news.mojom.h"
+#include "chrome/test/base/testing_profile.h"
+#include "content/public/test/browser_task_environment.h"
+#include "net/http/http_status_code.h"
+#include "net/traffic_annotation/network_traffic_annotation_test_helper.h"
+#include "services/data_decoder/public/cpp/test_support/in_process_data_decoder.h"
+#include "services/network/public/cpp/resource_request.h"
+#include "services/network/public/cpp/weak_wrapper_shared_url_loader_factory.h"
+#include "services/network/test/test_url_loader_factory.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace brave_news {
+
+constexpr char kPublishersResponse[] = R"([
+    {
+        "publisher_id": "111",
+        "publisher_name": "Test Publisher 1",
+        "feed_url": "https://tp1.example.com/feed",
+        "site_url": "https://tp1.example.com",
+        "category": "Tech",
+        "cover_url": "https://tp1.example.com/cover",
+        "cover_url": "https://tp1.example.com/favicon",
+        "background_color": "#FF0000",
+        "locales": [{
+          "locale": "en_NZ",
+          "channels": ["One", "Tech"]
+        }],
+        "enabled": false
+    }])";
+
+class BraveNewsInitializationPromiseTest : public testing::Test {
+ public:
+  BraveNewsInitializationPromiseTest()
+      : api_request_helper_(TRAFFIC_ANNOTATION_FOR_TESTS,
+                            test_url_loader_factory_.GetSafeWeakWrapper()),
+        pref_manager_(*profile_.GetPrefs()),
+        publishers_controller_(&api_request_helper_),
+        initialization_promise_(3, pref_manager_, publishers_controller_) {
+    // Ensure Brave News is enabled.
+    pref_manager_.SetConfig(mojom::Configuration::New(true, true, true));
+
+    // Disable backoffs, so we can test.
+    initialization_promise_.set_no_retry_delay_for_testing(true);
+  }
+  ~BraveNewsInitializationPromiseTest() override = default;
+
+ protected:
+  std::string InitializeAndGetLocale() {
+    base::RunLoop loop;
+    initialization_promise_.OnceInitialized(loop.QuitClosure());
+    loop.Run();
+
+    auto channels = pref_manager_.GetSubscriptions().channels();
+    if (channels.empty()) {
+      return "";
+    }
+    return channels.begin()->first;
+  }
+  std::string GetSourcesUrl() {
+    return "https://" + brave_news::GetHostname() + "/sources." +
+           brave_news::kRegionUrlPart + "json";
+  }
```

Comment:
nit: can this be a `static (constexpr)` method?

Reply:
Converted to `static` - It can't be `constexpr` because `GetHostname()` looks up runtime flags